modules/core/src/main/java/org/apache/synapse/config/xml/AbstractListMediatorFactory.java

```diff
@@ -34,18 +37,45 @@
 public abstract class AbstractListMediatorFactory extends AbstractMediatorFactory {
 
     protected static void addChildren(OMElement el, ListMediator m, Properties properties) {
-        Iterator it = el.getChildElements();
+        Iterator it = el.getChildren();
+
         while (it.hasNext()) {
-            OMElement child = (OMElement) it.next();
-            if (!DESCRIPTION_Q.equals(child.getQName())) { // neglect the description tag
-                Mediator med = MediatorFactoryFinder.getInstance().getMediator(child, properties);
-                if (med != null) {
-                    m.addChild(med);
-                } else {
-                    String msg = "Unknown mediator : " + child.getLocalName();
-                    log.error(msg);
-                    throw new SynapseException(msg);
+            OMNode child = (OMNode) it.next();
+            if (child instanceof OMElement) {
+                if (!DESCRIPTION_Q.equals(((OMElement) child).getQName())) { // neglect the description tag
+                    Mediator med = MediatorFactoryFinder.getInstance().getMediator((OMElement) child, properties);
+                    if (med != null) {
+                        m.addChild(med);
+                    } else {
+                        String msg = "Unknown mediator : " + ((OMElement) child).getLocalName();
+                        log.error(msg);
+                        throw new SynapseException(msg);
+                    }
                 }
+            } else if (child instanceof OMComment) {
+                CommentMediator commendMediator = new CommentMediator();
+                commendMediator.setCommentText(((OMComment) child).getValue());
+                m.addChild(commendMediator);
+            }
+        }
+    }
+
+    /**
+     * Find and add all comment nodes to the parent mediator as Comment Mediators
+     *
+     * @param el OMElement to extract OMComment Nodes
+     * @param m  Mediator to be updated with extracted OMComment nodes
+     */
+    protected static void addAllCommentChildrenToMediator(OMElement el, ListMediator m) {
+        Iterator it = el.getChildren();
```

Comment:
This particular iterator's scope is not properly restricted. Imagine someone using the same iterator instance somewhere down the code. If the same is modified within the while clause, then that can lead to inconsistent behaviours.